drivers/adc/adc_intel_quark_d2000.c

```diff
@@ -238,8 +238,8 @@ static int adc_quark_d2000_read_request(struct device *dev,
 
 	/* hardware requires minimum 10 us delay between consecutive samples */
 	if (seq_tbl->options &&
-	    seq_tbl->options->extra_samplings &&
-	    seq_tbl->options->interval_us < 10) {
+	    (seq_tbl->options->interval_us > 0) &&
```

Comment:
When there is only one sampling to be done, there is no point in checking the interval as it won't be used anyway. I guess that's why `extra_samplings` was checked here. And I guess this check could stay, only this proper handling of zero interval should be added (I mean `(seq_tbl->options->interval_us > 0)`).

Reply:
The whole if statement is incorrect. It was "kind of" inherited from previous adc_dw implementation before the ADC change. So these are no longer valid.